versions/development/SPEC.md

```diff
@@ -1218,81 +1259,343 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
+
+If the `disks` key is not defined then the engine must enforce a default value of `1 GiB` mounted at the execution directory. The engine must not provide a mechanism to set their own default value for disks which circumvents this,  ie `runtime-defaults`. 
+
+
+
+**Simple declaration**
+ 
+```wdl
+task disks_test {
+  #.....
+  runtime {
+    disks: 100
+  }
+}
+```
+
+**Declaration with mount point**
+```wdl
+task disks_test {
+  #.....
+  runtime {
+    disks: "/mnt/outputs 500 GiB"
   }
+}
+```
+ 
+ 
+**Declaration with multiple disks**
+```wdl
+task disks_test {
+  #.....
   runtime {
-    docker: "ubuntu:latest"
+  	# The first value will be mounted at the execution root
+    disks: ["500","/mnt/outputs 500 GiB","/mnt/tmp 5 TB"]
   }
 }
 ```
 
-#### memory
+#### maxRetries (**Optional**)
 
-Memory requirements for this task.  Two kinds of values are supported for this attributes:
+The `maxRetries` key provides a mechanism for a task to be retried in the event of a failure. If this key is defined, the engine must retry the task UP TO but not exceeding the number of attempts that it specifies.
 
-* `Int` - Interpreted as bytes
-* `String` - This should be a decimal value with suffixes like `B`, `KB`, `MB` or binary suffixes `KiB`, `MiB`.  For example: `6.2 GB`, `5MB`, `2GiB`.
+**Note**: The engine may choose to define an upper bound on the number of retry attempts which it permits. This number must exceed 1
+
+##### Default Value
+
+ If this key is not defined, then the engine must interpret the task as not retryable, therefore any failure in the task should never result in a retry by the engine, and the final status of the task should remain the same.
 
 ```wdl
-task memory_test {
-  input {
-    String arg
+task maxRetries_test {
+  #.....
+  runtime {
+    maxRetries: 4
   }
+}
+```
 
-  command {
-    python process.py ${arg}
+#### returnCodes (**Optional**)
+
+The `returnCodes` key provides a mechanism for a wdl to specify the return code which constitute a successful execution of a task.  The value of this key can be an, `Int`, `Array[Int]`, or `"*"`. The engine must honor the return codes specified within the runtime block and set the tasks status appropriately. 
+
+* `"*"` - This special value indicates that ALL returnCodes should be considered a success
+* `Int` - Only the specified return code should be considered a success
+* `Array[Int]` - Only the return codes specify in the array should be considered a success
+
+
+##### Default Value
+
+If no `returnCodes` key is provided, then a returnCode of `0` will be considered a success.
+
+**Single return code**
+```wdl
+task maxRetries_test {
+  #.....
+  runtime {
+    returnCodes: 1
+  }
+}
+```
+
+**Multiple return codes**
+```wdl
+task maxRetries_test {
+  #.....
+  runtime {
+    returnCodes: [1,2,5,10]
   }
+}
+```
+
+**All return codes**
+```wdl
+task maxRetries_test {
+  #.....
   runtime {
-    memory: "2GB"
+    returnCodes: "*"
   }
 }
 ```
 
+### Hints Section
```

Comment:
I know it's a bit late, but I'd like to suggest "reusable" as a reserved hint. Many execution engines have a concept of reusable results, i.e. for the exact same inputs on the exact same task, previously available outputs can be reused rather than re-running the task. The default value should be "true" but this will not be appropriate for a subset of tasks, and it may be desirable to (temporarily) turn this off for the entire workflow, e.g. when debugging.

Reply:
@jdidion unfortuinately no changes can be made to the current PR. Any changes we make need to be thropugh a separate PR process. If you would like to create a PR suggesting reusable as a runtime hint we would greatly welcome it!